modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1124,7 +1124,10 @@ protected CompletableFuture<IgniteBiTuple<ClusterNode, Long>> enlist(int partId,
         // TODO asch IGNITE-15091 fixme need to map to the same leaseholder.
         // TODO asch a leader race is possible when enlisting different keys from the same partition.
         return fut0.handle((primaryPeerAndTerm, e) -> {
-            if (primaryPeerAndTerm.get1() == null || e != null) {
+            if (e != null) {
```

Comment:
I split the block to 2 `if` blocks because the exception to be thrown is constructed a bit differently: in one case it has a cause, in the other it does not.